include/drivers/flash.h

```diff
@@ -296,6 +307,25 @@ static inline size_t z_impl_flash_get_write_block_size(struct device *dev)
 	return api->write_block_size;
 }
 
+
+/**
+ *  @brief  Get pointer to flash_parameters structure
+ *
+ *  @return pointer to flash_parameters structure characteristic for
+ *          the device.
+ */
+#if defined(CONFIG_FLASH_GET_PARAMETERS_API)
+__syscall const struct flash_parameters *flash_get_parameters(const struct device *dev);
+
+static inline const struct flash_parameters *z_impl_flash_get_parameters(const struct device *dev)
+{
+	const struct flash_driver_api *api =
+		(const struct flash_driver_api *)dev->driver_api;
+
+	return api->get_parameters();
```

Comment:
Because this API is optional, it'd be easy for out-of-tree drivers to not provide it, and never notice that somebody enabled the option.  There should at minimum be an `__ASSERT()` that the API field has been filled out, but I'd still rather see:

    return (api->get_parameters == NULL) ? NULL : api->get_parameters();

to at least let the application be able to recover in that situation.

Which then gets back to the question of whether `flash_get_parameters()` should always be available, and just return `NULL` if the option isn't enabled.  I don't recall what the resolution of that discussion was, if we reached one.

Reply:
I will make it permanent addition to API. I think it is not worth making this optional as checking whether this returns  NULL, after calls, is too much of a hassle.

Reply:
Even if turning this off saves us ~8 bytes of flash, the checking for NULL every time when the structure is requested would probably cost us as much more. And NULL processing at runtime is problematic because this is a thing that  is supposed to describe some property, what if it is not there? Stop running?